pilot/cmd/pilot-discovery/main.go

```diff
@@ -80,6 +85,15 @@ var (
 	}
 )
 
+func hasKubeRegistry() bool {
```

Comment:
Can you add a comment - no clue why having a kube registry affects the spiffe naming or trust domains. 

Is there any security reason ? Can we have a consistent security regardless of where the workloads run ?

Reply:
We did this to keep the previous behaviour with different defaults when running on K8s and elsewhere.